ssl/quic/quic_impl.c

```diff
@@ -2702,6 +2710,7 @@ int ossl_quic_set_incoming_stream_policy(SSL *s, int policy,
         break;
```

Comment:
There is an inconsistency in this function now (but github won't let me comment on the right line).

Above we call `expect_quic_conn_only`, but if that fails it calls `QUIC_RAISE_NON_NORMAL_ERROR`. Although I'm not sure if that is correct - because that has the effect of causing `SSL_get_error()` to return `SSL_ERROR_SSL` which should only really happen for "IO" functions (which this isn't).

Below we just call `ERR_raise`, but not `QUIC_RAISE_NON_NORMAL_ERROR`.

Reply:
If you don't want `last_error` changed for these calls, i would suggest the following changes:

- add a `is_io` argument to `quic_raise_non_normal_error` and make `last_error` setting conditional on this;
- add a new macro `QUIC_RAISE_NON_IO_ERROR` which is otherwise identical to `QUIC_RAISE_NON_NORMAL_ERROR`.

Error handling in the APL (`quic_impl.c`) should always be done in a principled fashion centralised through one of these macros.

Reply:
Implemented as Hugo suggested.